src/types.ts

```diff
@@ -48,7 +48,7 @@ export interface UnpluginOptions {
   buildEnd?: (this: UnpluginBuildContext) => Promise<void> | void
   transform?: (this: UnpluginBuildContext & UnpluginContext, code: string, id: string) => Thenable<TransformResult>
   load?: (this: UnpluginBuildContext & UnpluginContext, id: string) => Thenable<TransformResult>
-  resolveId?: (id: string, importer: string | undefined, options: { isEntry: boolean }) => Thenable<string | ExternalIdResult | null | undefined>
+  resolveId?: (this: UnpluginBuildContext & UnpluginContext, id: string, importer: string | undefined, options: { isEntry: boolean }) => Thenable<string | ExternalIdResult | null | undefined>
```

Comment:
Adding this would indicate that it works for **every bundler**. Would love to have a simple test for it.

Reply:
I added a simple test for this in `resolveId` (and perhaps we should in `transform` and `load` too).

And good that we tested: Webpack doesn't do this. Does anyone know whether `Resolver`s in Webpack can access a `compilation` object, so we can call `createContext`? I didn't see an easy way... One option would be to throw an error in these situations.

Reply:
I've done a bunch of searching and it seems like loaders and resolvers are treated very differently in Webpack, and only loaders have all the APIs we need. I'm unclear whether a loader can be used as a resolver. But at least for now, I've added the same consistent interface to `resolveId` (so types work and function calls don't crash from lack of functions), but most of the API functions `throw` errors. (Some do work; I could implement `error`, `warn`, and `parse` still.)

Also documented the limitations here (which are not new — previously `resolveId` offered *no* `this` API in Webpack or esbuild).